sonic-pi: use supercollider with sc3-plugins

lilyinstarlight opened the pull request:
###### Description of changes

Some synths in Sonic Pi, such as the `:piano` synth, require sc3-plugins to be loaded in SuperCollider. This PR uses the recently merged plugin support (#147011) to have SuperCollider automatically load sc3-plugins, as Sonic Pi expects. I additionally made a drive-by change to remove the `supercollider` package override, since the package is already using `fftwSinglePrec` anyways without it

See https://github.com/sonic-pi-net/sonic-pi/blob/v3.3.1/FAQ.md#linux-there-is-no-sound-with-use_synth-piano for info about sc3-plugins requirement

Also, I've been keeping up with the development of the imminent 4.0 release of Sonic Pi and have been working toward making it friendlier for Linux packaging. Relatedly, I've been keeping a Nix package up to date with recent working dev commits that I planned to PR to nixpkgs once 4.0 is stable (see [here](https://github.com/lilyinstarlight/foosteros/blob/main/pkgs/sonic-pi-beta/default.nix)). I'm happy to add myself to the maintainer list if you'd like, since I'm involved on the Sonic Pi core team and have been working on Nix packaging for Sonic Pi


###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

###### nixpkgs-review

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sonic-pi</li>
  </ul>
</details>

gador commented:
##### Reviewed points

- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] all depending packages build

##### Possible improvements

##### Comments

This is a semi-automatic executed nixpkgs-review with nixpkgs-review-checks extension. It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of nixpkgs-review pr 169851 run on x86_64-linux 1

<details>
<summary>
1 package built:
</summary>
<ul>
<li>sonic-pi</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
sonic-pi:
<details>

got build log for '/nix/store/dp3n8k3ryiz8n1qd0ckpzc5apnbh90wh-sonic-pi-3.3.1' from 'daemon'
When evaluating attribute ‘sonic-pi’:
warning: build-tools-in-build-inputs
pkg-config is a build tool so it likely goes to nativeBuildInputs, not buildInputs.

Near pkgs/applications/audio/sonic-pi/default.nix:55:3:

────────────────────
   |
55 |   buildInputs = [
   |   ^
────────────────────

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md

</details>
</details>

c0deaddict reviewed (approved):
LGTM!

c0deaddict commented:
> I'm happy to add myself to the maintainer list if you'd like, since I'm involved on the Sonic Pi core team and have been working on Nix packaging for Sonic Pi

Yes, please do, that would really help!

lilyinstarlight committed:
sonic-pi: use supercollider with sc3-plugins

Some synths in Sonic Pi, such as the `:piano` synth, require sc3-plugins
to be loaded in SuperCollider

Ref: https://github.com/sonic-pi-net/sonic-pi/blob/dev/FAQ.md#linux-there-is-no-sound-with-use_synth-piano

lilyinstarlight commented:
> Yes, please do, that would really help!

Done! I'll probably have more PRs later backporting some of the improvements from my development package (like having a .desktop file, stripping the `app` directory, or the `{,native}buildInputs` QA issue in the nixpkgs-review comment above)

SuperSandro2000 commented on pkgs/applications/audio/sonic-pi/default.nix:
```diff
@@ -14,14 +13,12 @@
 }:
 
 let
 
```
```suggestion
```

SuperSandro2000 committed:
Update pkgs/applications/audio/sonic-pi/default.nix
